### PR TITLE
[FLINK-30027] Fields min and max in BinaryTableStats support lazy deserialization

### PR DESCRIPTION
**Pull request by zjureel**
`Predicate` get min and max from `BinaryRowData`, lazily deserialization.

### CODE REVIEW

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/file/stats/BinaryTableStats.java [hunk 1]
```diff
@@ -73,27 +74,40 @@ public FieldStats[] fields(FieldStatsArraySerializer converter, @Nullable Long r
...
 
     public RowData toRowData() {
-        return GenericRowData.of(
-                serializeBinaryRow(min), serializeBinaryRow(max), new GenericArrayData(nullCounts));
+        return row == null
+                ? GenericRowData.of(
+                        serializeBinaryRow(min()),
+                        serializeBinaryRow(max()),
+                        new GenericArrayData(nullCounts()))
+                : GenericRowData.of(row.getBinary(0), row.getBinary(1), row.getArray(2));
```
Comment (JingsongLi) [inline flink-table-store-core/src/main/java/org/apache/flink/table/store/file/stats/BinaryTableStats.java:106]: Why not just return row?

**Review by JingsongLi — COMMENTED** [review]
Thanks @zjureel , can you add a test for three constructors?

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/file/stats/BinaryTableStats.java (hunk 1 above)
Comment (zjureel) [reply flink-table-store-core/src/main/java/org/apache/flink/table/store/file/stats/BinaryTableStats.java]: Done

**Comment by zjureel** [comment]
@JingsongLi Done

**Review by JingsongLi — APPROVED** [review]
Looks good to me!

**Pull request merged**